compiler/src/dotty/tools/io/ZipArchive.scala

```diff
@@ -274,12 +206,11 @@ final class ManifestResources(val url: URL) extends ZipArchive(null) {
     for (zipEntry <- iter) {
       val dir = getDir(dirs, zipEntry)
       if (!zipEntry.isDirectory) {
-        class FileEntry() extends Entry(zipEntry.getName) {
+        val f = new Entry(zipEntry.getName) {
```

Comment:
@odersky - despite #2282 and other fixes - this still had to be changed in order for Dotty to accept it. I've had a hard time minimizing it, but I'll take another crack at it.